rust/main/utils/run-locally/src/starknet/mod.rs

```diff
@@ -0,0 +1,570 @@
+                .map(|v| format!("0x{:x}", v))
+                .collect::<Vec<String>>();
+
+            let fee_amount = 0u32;
+
+            let initial_args = vec![
+                target.domain.to_string(),
+                format!("u256:{}", target.deployments.mock_receiver.clone()),
+                strk_msg_len.to_string(),
+                strk_msg_str.len().to_string(),
+            ];
+
+            // we set the options to `None` for now
+            // which means no hook nor hook_metadata
+            let options_args = vec!["1".to_string(), "1".to_string()];
+
+            let args = initial_args
+                .into_iter()
+                .chain(strk_msg_str)
+                .chain(vec![format!("u256:{}", fee_amount)])
+                .chain(options_args)
+                .collect();
+
+            cli.send_tx(
+                node.deployments.mailbox.clone(),
+                "dispatch".to_string(),
+                args,
+            );
+        }
+    }
+
+    let _stack = StarknetHyperlaneStack {
+        validators: hpl_val.into_iter().map(|v| v.join()).collect(),
+        relayer: hpl_rly.join(),
+    };
+
+    // Mostly copy-pasta from `rust/utils/run-locally/src/main.rs`
+    let loop_start = Instant::now();
+    let mut failure_occurred = false;
+    loop {
+        // look for the end condition.
+        if termination_invariants_met(
+            hpl_rly_metrics_port,
+            dispatched_messages,
+            starting_relayer_balance,
+        )
+        .unwrap_or(false)
+        {
+            // end condition reached successfully
+            break;
+        } else if (Instant::now() - loop_start).as_secs() > TIMEOUT_SECS {
+            // we ran out of time
+            log!("timeout reached before message submission was confirmed");
+            failure_occurred = true;
+            break;
+        }
+
+        sleep(Duration::from_secs(5));
+    }
+
+    if failure_occurred {
+        panic!("E2E tests failed");
+    } else {
+        log!("E2E tests passed");
+    }
+}
+
+fn termination_invariants_met(
+    relayer_metrics_port: u32,
+    messages_expected: u32,
+    _starting_relayer_balance: f64,
+) -> eyre::Result<bool> {
+    // Commented as IGP is not implemented for Starknet
+    // let gas_payments_scraped = fetch_metric(
+    //     &relayer_metrics_port.to_string(),
+    //     "hyperlane_contract_sync_stored_events",
+    //     &hashmap! {"data_type" => "gas_payment"},
+    // )?
+    // .iter()
+    // .sum::<u32>();
+    // let expected_gas_payments = messages_expected;
+    // if gas_payments_scraped != expected_gas_payments {
+    //     log!(
+    //         "Relayer has indexed {} gas payments, expected {}",
+    //         gas_payments_scraped,
+    //         expected_gas_payments
+    //     );
+    //     return Ok(false);
+    // }
+
+    let delivered_messages_scraped = fetch_metric(
+        &relayer_metrics_port.to_string(),
+        "hyperlane_operations_processed_count",
```

Comment:
I wonder why we don't use `hyperlane_messages_processed_count` metric as Ethereum+Sealevel E2E tests use?

Reply:
we do the same as for cosmos so we use `hyperlane_operations_processed_count`